src/valkey-cli.c

```diff
@@ -2397,6 +2425,25 @@ static int cliSendCommand(int argc, char **argv, long repeat) {
             fflush(stdout);
             if (config.pubsub_mode || num_expected_pubsub_push > 0) {
                 if (isPubsubPush(config.last_reply)) {
+                    /* Handle pubsub mode */
+                    if (config.last_reply->elements >= 3) {
+                        char *cmd = config.last_reply->element[0]->str;
+
+                        /* If it's an unsubscribe command, call the helper */
+                        if (strcmp(cmd, "unsubscribe") == 0 || strcmp(cmd, "punsubscribe") == 0 || strcmp(cmd, "sunsubscribe") == 0) {
+                            if (config.pubsub_mode && isPubsubPush(config.last_reply)) {
+                                processUnsubscribeReply(config.last_reply);
+                            }
+                        } 
+
+                        /* Handle subscribe commands */
+                        else if (strcmp(cmd, "subscribe") == 0 || strcmp(cmd, "psubscribe") == 0) {
+                            config.pubsub_unsharded_count++;
```

Comment:
Just noticed: In the unsubscribe case, you just set `config.pubsub_unsharded_count = reply->element[2]->integer`. We should do the same here.

The corner case where this matters is when the user subscribes to the same channel twice:

```
127.0.0.1:6379> subscribe x
1) "subscribe"
2) "x"
3) (integer) 1
127.0.0.1:6379(subscribed mode)>  subscribe x
1) "subscribe"
2) "x"
3) (integer) 1
```